src/components/TradeWidget/index.tsx

```diff
@@ -356,6 +356,8 @@ const TradeWidget: React.FC = () => {
 
   // If user is connected, use balances, otherwise get the default list
   const tokens = useMemo(
+    // it's okay to tokenListApi.getTokens() here without subscribing to updates
+    // because balances from useTokenBalances is already subscribed
     () => (isConnected && balances.length > 0 ? balances : tokenListApi.getTokens(fallBackNetworkId)),
```

Comment:
But, balances is only used/updated if wallet is connected.
Won't we allow tokens to be added on trade page without having connected the wallet?

I guess that is fine for adding tokens, but can be a problem due to the way I'll load the tokens. (can probably address on my PR, though)

Reply:
ugh, you are right
having default network-less tokenList is a problem